### ISSUE TITLE
docs-bug(icon): `MatIcon` appears in the `Directives` section

### ISSUE THREAD

**Issue by dev054** [angular/components#24093]
### Documentation Feedback

Currently the `MatIcon` appears in the `Directives` section, however it's a `Component`:

<img width="255" alt="MatIcon in Directives" src="https://user-images.githubusercontent.com/28629080/145718535-b16b9676-6455-4300-aff5-31b572114235.png">

### Affected documentation page

https://material.angular.io/components/icon/api

**Comment by dev054** [comment]
Oh, I just noticed it's happening on all pages, not just for the `MatIcon`, so now I'm starting to wonder if this is on purpose 🤔

**Comment by crisbeto** [comment]
It's a bit of trivia from the Angular.js days, but every component is also a directive.

**Comment by dev054** [comment]
Yes, I'm aware that a component is, under the hood, a directive, but shouldn't the docs differentiate them, like the Angular docs?

**Comment by mistrykaran91** [comment]
@crisbeto : I have created MR for `autocomplete` just to be sure whether its a correct approach and if that's the case then I can create another MR updating all the other components.

https://github.com/angular/material2-docs-content/pull/26

**Comment by crisbeto** [comment]
@mistrykaran91 the content in that repo is generated from source code. Your change will be overwritten next time we merge a PR.

**Comment by mistrykaran91** [comment]
My bad, I'll check the other way round.

**Comment by mistrykaran91** [comment]
So, its the bazel tools project which is used to generate those documentation ?

**Comment by mistrykaran91** [comment]
<img width="484" alt="Image" src="https://github.com/user-attachments/assets/fee4893b-29d8-45ec-a56c-3acde56e8cbc" />

@crisbeto : It seems we have some property names which are prefixed with `directive` so should 

1. I rename to something else which serves both the purpose like (`directiveOrComponentExportAs`, etc )
2. I use the same names here with adding only one extra property i.e, `isComponent` specifically as both of them have all other properties which are same.
3. I create new property kind of repeated for component ?

**Comment by mistrykaran91** [comment]
@crisbeto : Any documentation or `README.md` just to test the documentation with my changes ?

**Comment by mistrykaran91** [comment]
@crisbeto @andrewseguin: I have created this MR - https://github.com/angular/components/pull/30661 to fix this issue.
